usr/src/uts/common/fs/zfs/zfs_vnops.c

```diff
@@ -4424,6 +4429,66 @@ zfs_inactive(vnode_t *vp, cred_t *cr, caller_context_t *ct)
 	rw_exit(&zfsvfs->z_teardown_inactive_lock);
 }
 
+static void
+zfs_inactive_task(void *task_arg)
+{
+	znode_t *zp = task_arg;
+	ASSERT(zp);
+	zfs_inactive_impl(zp, CRED());
```

Comment:
I believe using CRED() in a taskq task is ok since we already do that in vn_rele_inactive()